### PR TITLE
feat: Enable custom check prefixes

### LINKED ISSUE #8: Custom assertion prefixes
We currently use `→` to indicate whether a comment is a connoisseur test. I think users should be able to replace it with a custom character or a string.

### PR DESCRIPTION
**Pull request by onurtemizkan**
Fixes #8.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,8 +5,26 @@
 const connoisseur = require("./lib/connoisseur");
 const fs = require("fs");
 const concat = require("concat-stream");
-const argv = require("yargs").boolean("f").argv;
+const argv = require("yargs")
+                .options({
```
Comment (BYK) [inline index.js]: Oh boy why so much indent? :(

File: index.js [hunk 2]
```diff
@@ -5,8 +5,26 @@
...
+                    /* eslint-disable id-length */
+                    f: {
+                        alias: "file",
+                        describe: "Use file as input",
+                        type: "boolean",
+                        default: false,
+                    },
+                    p: {
+                        alias: "commentPrefix",
```
Comment (BYK) [inline index.js]: No camelCase in command line args. Should be `comment-prefix`. I may prefer `check-prefix` too.

File: test/connoisseur.spec.js [hunk 3]
```diff
@@ -11,27 +11,27 @@ const connoisseur = require(path.join(__dirname, "..", "lib", "connoisseur.js"))
 
 describe("lib", () => {
     it("should return an empty array for an empty string", () => {
-        expect(connoisseur("")).to.be.empty();
+        expect(connoisseur("", "STDIN", { commentPrefix: " → " })).to.be.empty();
```
Comment (BYK) [inline test/connoisseur.spec.js]: I don't we should include spacing in the option.

**Review by BYK — CHANGES_REQUESTED** [review]
Good job. Just a few comments. Major concern is camelCase in CLI arguments.

**Comment by onurtemizkan** [comment]
Updated PR 😇

File: test/connoisseur.spec.js [hunk 4]
```diff
@@ -38,65 +38,65 @@ describe("connoisseur", () => {
...
-            it("should print no error message for plain javascript ", done => {
-                const proc = spawn("node", ["./index.js"]);
-                proc.stdin.write(`var a = "foo"${EOL}`);
-                proc.stdout.on("data", data => {
-                    expect(data.toString()).to.be.equal("No errors found.\n");
-                    done();
-                });
-                proc.stdin.end();
+    describe("stdin", () => {
+        it("should print no error message for plain javascript ", done => {
```
Comment (BYK) [inline test/connoisseur.spec.js]:
```suggestion
        it("should print no error message for plain JavaScript", done => {
```

File: test/connoisseur.spec.js [hunk 5]
```diff
@@ -38,65 +38,65 @@ describe("connoisseur", () => {
...
 
-            it("should print no error message for a valid javascript source with a valid connoisseur test", done => {
-                const proc = spawn("node", ["./index.js"]);
-                proc.stdin.write("[4, 9, 16].map(Math.sqrt);  // → [ 2, 3, 4 ]");
-                proc.stdout.on("data", data => {
-                    expect(data.toString()).to.be.equal("No errors found.\n");
-                    done();
-                });
-                proc.stdin.end();
+        it("should print no error message for a valid javascript source with a valid connoisseur test", done => {
```
Comment (BYK) [inline test/connoisseur.spec.js]:
```suggestion
        it("should print no error message for a valid JavaScript source with a valid connoisseur test", done => {
```

File: test/connoisseur.spec.js [hunk 6]
```diff
@@ -38,65 +38,65 @@ describe("connoisseur", () => {
...
-                const proc = spawn("node", ["./index.js"]);
-                proc.stdin.write("[4, 9, 16].map(Math.sqrt);  // → [ 1, 2, 3 ]");
-                proc.stdout.on("data", data => {
-                    expect(data.toString()).to.be.equal(
-                        "Errors found for file STDIN:\n\t[4, 9, 16].map(Math.sqrt);\t→ Expected: [ 1, 2, 3 ], Found: [ 2, 3, 4 ]\n\n"
-                    );
-                    done();
-                });
-                proc.stdin.end();
+        it("should print errors for a valid javascript source with an invalid connoisseur test", done => {
```
Comment (BYK) [inline test/connoisseur.spec.js]:
```suggestion
        it("should print errors for a valid JavaScript source with an invalid connoisseur test", done => {
```

File: test/connoisseur.spec.js [hunk 7]
```diff
@@ -38,65 +38,65 @@ describe("connoisseur", () => {
...
+                done();
             });
+            proc.stdin.end();
         });
     });
 
     describe("lib", () => {
         it("should return an empty array for an empty string", () => {
-            expect(connoisseur("")).to.be.empty();
+            expect(connoisseur("", "STDIN", { checkPrefix: "→" })).to.be.empty();
```
Comment (BYK) [inline test/connoisseur.spec.js:79]: Shouldn't we have a default for this so the old API works without extra config?

**Review by BYK — APPROVED** [review]
Looks pretty good, thanks a lot!

**Pull request merged**